### PR TITLE
Nasindam/edge swagger spec

### PR DESCRIPTION
**Pull request by sindamnataraj**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x]  [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+            "default": {
+              "description": "Error response describing why the operation failed.",
+              "schema": {
+                "$ref": "#/definitions/ErrorResponse"
+              }
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextLink"
+          }
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:105]: There shouldn't be paging on single GET. (Copy paste from list call?)

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+              "$ref": "#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "$ref": "#/parameters/EdgeSubscriptionNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:172]: >responses [](start = 11, length = 9)

Should also include the default response like on the other operations.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/DeleteEdgeSub.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "parameters": {
+    "subscriptionId": "dd8597b4-8739-4467-8b10-f8679f62bfbf",
+    "resourceGroup": "testResourceGroup",
+    "edgeSubscriptionName": "c581d377-070e-4f7c-90f0-e06ea8dca9aa",
+    "api-version": "2020-06-01-preview"
+  },
+  "responses": {
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/DeleteEdgeSub.json:8]: >responses [](start = 3, length = 9)

Missing 204 response in the example.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+      },
+      "DeviceIdParameter": {
+        "description": "Device Id",
+        "name": "deviceId",
+        "type": "string",
+        "in": "query",
+        "required": false,
+        "x-ms-parameter-location": "method"
+      },
+      "EdgeSubscriptionParameter": {
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:367]: >EdgeSubscriptionParameter [](start = 7, length = 25)

This doesn't seem to be used.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+            "Create or update an Azure Stack registration.": {
+              "$ref": "examples/EdgeSubscription/PutEdgeSub.json"
+            }
+          },
+          "tags": [
+            "EdgeSubscriptions"
+          ],
+          "description": "Create a new edge subscription",
+          "operationId": "EdgeSubscriptions_Create",
+          "parameters": [
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:158]: >parameters [](start = 11, length = 10)

There doesn't seem to be a request body parameter.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/PutEdgeSub.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      "properties": {
+        "remoteSubscriptionId": "9e9704ab-561a-4498-ac60-64b2314456ee"
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "subscriptionId": "dd8597b4-8739-4467-8b10-f8679f62bfbf",
+        "resourceGroup": "testResourceGroup",
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/PutEdgeSub.json:17]: These are not valid top-level resource properties.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/dd8597b4-8739-4467-8b10-f8679f62bfbf/resourceGroups/testResourceGroup/providers/Microsoft.AzureStack/edgeSubscriptions/c581d377-070e-4f7c-90f0-e06ea8dca9aa",
+            "name": "c581d377-070e-4f7c-90f0-e06ea8dca9aa",
+            "type": "Microsoft.AzureStack/edgeSubscriptions",
+            "subscriptionId": "dd8597b4-8739-4467-8b10-f8679f62bfbf",
+            "resourceGroup": "testResourceGroup",
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json:16]: These are not valid top-level resource properties.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            "location": "southcentralus",
+            "entityTag": "02009249-0000-0000-0000-592854110000",
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json:18]: >"entityTag": "02009249-0000-0000-0000-592854110000", [](start = 12, length = 52)

This is not a valid top-level resource properties.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json (hunk 7 above)
Comment (majastrz) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListEdgeSubs.json:16]: This also applies to the other examples.

---
In reply to: [439719864](https://github.com/Azure/azure-rest-api-specs/pull/9807#discussion_r439719864) [](ancestors = 439719864)

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListDeviceIdSubs.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
+{
+  "parameters": {
```
Comment (majastrz) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/ListDeviceIdSubs.json:2]: >parameters [](start = 3, length = 10)

The examples seems like a duplicate of the other list example and it's not referenced in the swagger.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/DeleteEdgeSub.json (hunk 3 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/examples/EdgeSubscription/DeleteEdgeSub.json:8]: Added 204 response.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 4 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json]: Removed the unused parameter.

**Comment by majastrz** [comment]
@chiragg4u Can you take over the ARM review here?

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 1 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:102]: Likely got copied, removed it now.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 5 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:155]: Added the request body param NewEdgeSubscriptionParameter

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 2 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:172]: Added the default response.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+            {
+              "$ref": "#/parameters/EdgeSubscriptionNameParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK"
```
Comment (markcowl) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:131]: same comment on response descriptions - how does 200 differ from 204 for this operation?

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+              "$ref": "#/parameters/NewEdgeSubscriptionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK",
+              "schema": {
+                "$ref": "#/definitions/EdgeSubscription"
+              }
+            },
```
Comment (markcowl) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:178]: same as above - need an operation-specific description for status codes, this applies throughout

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 12]
```diff
@@ -0,0 +1,403 @@
...
+            "readOnly": true
+          },
+          "target": {
+            "description": "The target of the particular error.",
+            "type": "string",
+            "readOnly": true
+          }
+        }
+      },
+      "ErrorResponse": {
```
Comment (markcowl) [inline specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:267]: Please use the default ARM error pattern, as done here: https://github.com/Azure/azure-rest-api-specs/blob/d77baea351ade4453199f73e9aad1f4b8a320a76/specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json#L4418

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 1 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:102]: I have responded to your comments, can you please take a look? @majastrz

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 12 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:264]: Updated.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 11 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json:175]: Updated.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json (hunk 10 above)
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json]: Updated.

File: specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json [hunk 13]
```diff
@@ -0,0 +1,398 @@
...
+            {
+              "$ref": "#/parameters/DeviceIdParameter"
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "OK",
```
[the remark this answers is not in the document]
Comment (sindamnataraj) [reply specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/EdgeSubscription.json]: Updated.

**Review by chiragg4u — CHANGES_REQUESTED** [review]
I'm not sure we close the design on how Edge subscriptions are being exposed. Please share the details if this design was green lighted?
Also if we are still going with this for intermediate steps I would not like this to be publish it in public swagger repo as it will cause issues when we remove this later.

**Comment by markcowl** [comment]
@sindamnataraj Are we still intending to get this reviewd, or can we close?  Will close if there are no comments in the next 10 days.

**Review by markcowl — APPROVED** [review]

**Review by markcowl — CHANGES_REQUESTED** [review]
Need to fix required swagger chanks, please see the logs for details.

**Comment by markcowl** [comment]
Closing, per previous comment

**Pull request closed without merging**